### DISCUSSION TITLE
How to use two different views with interia and vuejs

### DISCUSSION THREAD

**Discussion by Maninder-cell** [inertiajs/inertia#772] [Help (Vue)]
in interia we have the only view which is default as app.blade.php. Is there any way that I can use another view like seller.blade.php and admin.blade.php. i know it would break the rule of single page application. but if you have any idea please let me know. i really need help with it. because I have different templates for different sections of the website.

**Comment by xavi7th** [comment]
In HandleInertiaRequests middleware, you can overwrite the rootview method and implement the logic required to determine the view that should be returned for a particular request. I am currently in transit and writing from my phone I would have written some examples.

**Reply by Maninder-cell** [reply]
can you please show me some of your examples? i really need help with it.

**Reply by Maninder-cell** [reply]
@xavi7th please reply i need help with it

**Reply by xavi7th** [reply]
@Maninder-cell I am sorry for the slow response. Here are a couple of ways I have handled this

```
<?php

namespace App\Http\Middleware;

use Inertia\Middleware;
// ... other uses

class HandleInertiaRequests extends Middleware
{
  protected $rootView = 'publicpages::app';

  public function share(Request $request)
  {
    // stuffs here
  }

  public function rootView(Request $request):string
  {
     if ($request->user()) {
    // This returns the relevant namespaced app.blade.php file, ex appuser::app, admin::app, accountant::app
      return strtolower($request->user()->getType()) . '::app';
    } elseif (Str::contains(\Route::currentRouteName(), 'auth')) {
      return 'userauth::app';
    } else {
      return $this->rootView;
    }
  }
}
```


```
/**
   * Sets the root template that's loaded on the first page visit.
   *
   * @see https://inertiajs.com/server-side-setup#root-template
   * @param Request $request
   * @return string
   */
  public function rootView(Request $request):string
  {
    if (Str::containsAll(\Route::currentRouteName(), ['super', 'admin', 'login'])) {
      return 'superadmin::app';
    } elseif (Str::containsAll(\Route::currentRouteName(), ['admin', 'login'])) {
      return 'admin::app';
    } elseif (Str::contains(\Route::currentRouteName(), ['create_account', 'login', 'password_reset'])) {
      return 'appuser::app';
    } else {
      return $this->rootView;
    }
  }
```


```
  public function rootView(Request $request):string
  {
   // This project doesn't use modules so I have different "mounting" blade files in the views directory  
   // and load the appropriate "app" blade file based on the login user.
    if ($request->user()->isAdmin()) {
      return 'superadmin-app';
    } elseif ($request->user()->isAcountant()) {
      return 'accountant-app';
    } else {
      return 'app';
    }
  }
```

I hope this helps.

**Comment by Maninder-cell** [comment]
@xavi7th have you call this middleware in routes or not because I know inertia middleware only runs when we refresh the page but I don't want that. i want root view changes without refreshing the page would it work.

**Reply by Maninder-cell** [reply]
@xavi7th and one thing how you have implemented this. why this middleware has the same three function rootView again and again.  I'm pretty confused can you explain it.

**Reply by xavi7th** [reply]
@Maninder-cell Inertia middleware runs on all requests, even Inertia requests. You can verify that by viewing the page response and checking to see that the "share()" object is always present in your response.

Sorry about the confusion. I was sharing the rootView() method from 3 different projects just to show you different examples of how I had implemented the multiple views solution.

**Reply by xavi7th** [reply]
@Maninder-cell and no I do not implement it in routes ( I guess you can if you rally need to) but I go with the docs and just apply it to the web route group.

```
Next, setup the Inertia middleware. In the Rails adapter, this is configured automatically for you. However, in Laravel you need to publish the HandleInertiaRequests middleware to your application, which can be done using this artisan command:

php artisan inertia:middleware
Once generated, register the HandleInertiaRequests middleware in your App\Http\Kernel, as the last item in your web middleware group.

'web' => [
    // ...
    \App\Http\Middleware\HandleInertiaRequests::class,
],
```
